sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaSourceDescriptor.java

```diff
@@ -50,6 +50,14 @@
   @Nullable
```

Comment:
It seems like it should be an error to specify both `stop_read_time` and `stop_read_offset`, similarly for `start_read_time` and `start_read_offset`.

We can add some simple error checking within the static method `of(...)` and `create(...)`